src/main/java/com/minecolonies/util/StructureWrapper.java

```diff
@@ -275,143 +233,45 @@ private void placeBlock(IBlockState localState, @NotNull Block localBlock, @NotN
     }
 
     /**
-     * Scan the schematic and save it to the disk.
+     * Scan the structure and save it to the disk.
      *
      * @param world Current world.
      * @param from  First corner.
      * @param to    Second corner.
      * @return Message to display to the player.
      */
-    public static String saveSchematic(@Nullable World world, @Nullable BlockPos from, @Nullable BlockPos to)
+    public static String saveStructure(@Nullable World world, @Nullable BlockPos from, @Nullable BlockPos to)
     {
         if (world == null || from == null || to == null)
         {
             throw new IllegalArgumentException("Invalid method call, arguments can't be null. Contact a developer.");
         }
 
-        @NotNull Schematic schematic = scanSchematic(world, from, to);
+        BlockPos blockpos =
+                new BlockPos(Math.min(from.getX(), to.getX()), Math.min(from.getY(), to.getY()), Math.min(from.getZ(), to.getZ()));
+        BlockPos blockpos1 =
+                new BlockPos(Math.max(from.getX(), to.getX()), Math.max(from.getY(), to.getY()), Math.max(from.getZ(), to.getZ()));
+        BlockPos size = blockpos1.subtract(blockpos).add(1, 1, 1);
 
-        String fileName = LanguageHandler.format("item.scepterSteel.scanFormat", schematic.getType(), System.currentTimeMillis());
-        @NotNull File file = new File(getScanDirectory(world), fileName);
+        WorldServer worldserver = (WorldServer) world;
+        MinecraftServer minecraftserver = world.getMinecraftServer();
+        TemplateManager templatemanager = worldserver.getStructureTemplateManager();
 
-        if (SchematicFormat.writeToFile(file, schematic))
+        // if the directory does not exist, create it
+        createScanDirectory(world);
+        String currentMillis = Long.toString(System.currentTimeMillis());
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Local variable 'currentMillis' could be declared final [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=pmd%3ALocalVariableCouldBeFinal)